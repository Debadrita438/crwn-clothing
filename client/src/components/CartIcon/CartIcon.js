import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { toggleCartHidden } from '../../redux/cart/cartAction';
import { selectCartItemsCount } from '../../redux/cart/cartSelector';

import { CartIconContainer, ItemCountStyles, ShoppingIcon } from './CartIcon.styles';

const mapDispatchToProps = dispatch => ({
    toggleCartHidden: () => dispatch(toggleCartHidden())
});

const mapStateToProps = createStructuredSelector({
    itemCount: selectCartItemsCount
});

export const CartIcon = ({ toggleCartHidden, itemCount }) => (
    <CartIconContainer onClick={ toggleCartHidden }>
        <ShoppingIcon />
        <ItemCountStyles>{ itemCount }</ItemCountStyles>
    </CartIconContainer>
);

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);