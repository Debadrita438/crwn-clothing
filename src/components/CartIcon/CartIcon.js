import { ReactComponent as ShoppingIcon } from '../../assets/shopping-bag.svg';
import { toggleCartHidden } from '../../redux/cart/cartAction';

import { connect } from 'react-redux';

import './CartIcon.scss';

const mapDispatchToProps = dispatch => ({
    toggleCartHidden: () => dispatch(toggleCartHidden())
})

const CartIcon = ({ toggleCartHidden }) => (
    <div className='cart-icon' onClick={ toggleCartHidden }>
        <ShoppingIcon className='shopping-icon' />
        <span className="item-count">0</span>
    </div>
);

export default connect(null, mapDispatchToProps)(CartIcon);