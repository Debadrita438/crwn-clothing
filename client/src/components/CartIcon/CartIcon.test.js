import { shallow } from 'enzyme';

import { CartIcon } from './CartIcon';

describe('CartIcon Component', () => {
    let wrapper;
    let mockToggleCartHidden
    let mockItemCount = 5;

    beforeEach(() => {
        mockToggleCartHidden = jest.fn();

        const mockProps = {
            toggleCartHidden: mockToggleCartHidden,
            itemCount: mockItemCount
        };

        wrapper = shallow(<CartIcon { ...mockProps } />);
    });

    it('should render CartIcon component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call toggleCartHidden when clicked', () => {
        wrapper.find('CartIconContainer').simulate('click');
        expect(mockToggleCartHidden).toHaveBeenCalled();
    });

    it('should show the itemCount as the text', () => {
        const itemCount = parseInt(wrapper.find('ItemCountStyles').text());
        expect(itemCount).toBe(5);
    });
});