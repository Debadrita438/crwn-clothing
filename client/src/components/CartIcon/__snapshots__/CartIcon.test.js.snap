// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon Component should render CartIcon component 1`] = `ShallowWrapper {}`;
